### DISCUSSION TITLE
[Proposal]: Inferred enum values

### DISCUSSION THREAD

**Discussion by Jijjy** [dotnet/csharplang#5958] [Language Ideas] [answered]
Say for example I have a "Direction" enum
``` cs
enum Direction
{
  None = 0,
  North = 1,
  South = 2,
  East = 4,
  West = 8
}
```

And I have a variable `var foo = Direction.North | Direction.East;`
It would be great if you could e.g. call HasFlag without having to specify the enum
e.g. `if(foo.HasFlag(North))` instead of `if(foo.HasFlag(Direction.North))`

You could also write something like `Direction foo = North | East;`

In this instance, having "Direction" isn't too bad  but in many cases enums can have long names or be nested inside classes which can hurt readability.

If a local variable is declared e.g. `North` in this instance, that should take precedence. If for some reason the enum type cannot be inferred or is ambiguous, then that should result in a compilation error.

If this was added, I think it may sometimes be unclear where the values come from, however if they are highlighted like enum types then I think it should be obvious that they're not local variables or something.

**Comment by FaustVX** [comment, marked as answer]
Already discussed in #2926

**Reply by jnm2** [reply]
And championed! 🎉

**Reply by FaustVX** [reply]
When something is championed, what does that mean ?
You actively working on it ?
It will be ready for the next version of C# ?

**Reply by HaloFour** [reply]
"Championed" means that a member of the language design team is willing to bring it to a language design meeting and discuss it, and to marshal it through the rest of the design process.  It doesn't provide any level of guarantee as to when the feature will ship, if at all.

**Reply by FaustVX** [reply]
Ok, thanks.
